R/board_azure.R

```diff
@@ -20,31 +26,59 @@
 #' \dontrun{
 #' # To create a board that you can write to, you'll need to supply one
 #' # of `key`, `token`, or `sas` to AzureStor::blob_container()
-#' container <- AzureStor::blob_container(url, key = "my-key")
-#' board <- board_azure(container)
+#' # First, we create a board using the classic Azure blob API
+#' url <- "https://myaccount.blob.core.windows.net/mycontainer"
+#' container <- AzureStor::blob_container(url, sas = "my-sas")
+#' board <- board_azure(container, "path/to/board")
 #' board %>% pin_write(iris)
+#'
+#' # ADLSgen2 is a modern, efficient way to access blobs
+#' # - Use 'dfs' instead of 'blob' in the account URL to use the ADLSgen2 API
+#' # - We reuse the board created via the blob API above
+#' adls_url <- "https://myaccount.dfs.core.windows.net/mycontainer"
+#' container <- AzureStor::storage_container(adls_url, sas = "my-sas")
+#' board <- board_azure(container, "path/to/board")
+#' board %>% pin_list()
+#' board %>% pin_read("iris")
 #' }
-board_azure <- function(container, n_processes = 10, versioned = TRUE, cache = NULL) {
+board_azure <- function(container, path = "", n_processes = 10, versioned = TRUE, cache = NULL) {
```

Comment:
I think you need to document `path` since it's inheriting a description that doesn't make sense here.

And maybe it should be called `prefix` to match `board_s3()`?

Reply:
This is documented up in line 8. It's `path` by analogy to the path for a local board. "Prefix" is an S3-ism which was copied by blob storage, but it doesn't really apply for the other storage services. I'll clarify this.

Reply:
Ok, I missed that. I think the explanation and variable name makes sense.